lib/particular-calendars/canada.ts

```diff
@@ -1,88 +1,119 @@
 import { PatronTitles } from '../constants/martyrology-metadata';
 import { Precedences } from '../constants/precedences';
 import { CalendarDef } from '../models/calendar-def';
-import { Inputs } from '../types/calendar-def';
+import { Inputs, ParticularConfig } from '../types/calendar-def';
 import { Americas } from './americas';
 
 export class Canada extends CalendarDef {
   parentCalendar = Americas;
 
+  particularConfig: ParticularConfig = {
+    epiphanyOnSunday: true,
+    ascensionOnSunday: true,
+    corpusChristiOnSunday: true,
+  };
+
   inputs: Inputs = {
+    // src: mr_fr_2021_ed3
     andre_bessette_religious: {
       precedence: Precedences.ProperMemorial_11b,
       dateDef: { month: 1, date: 7 },
     },
 
+    // src: mr_fr_2021_ed3
     raymond_of_penyafort_priest: {
-      precedence: Precedences.OptionalMemorial_12,
       dateDef: { month: 1, date: 8 },
     },
 
+    // src: mr_fr_2021_ed3
     marguerite_bourgeoys_virgin: {
       precedence: Precedences.ProperMemorial_11b,
       dateDef: { month: 1, date: 12 },
     },
 
+    // src: mr_fr_2021_ed3
     joseph_spouse_of_mary: {
       customLocaleId: 'joseph_spouse_of_mary_patron_of_canada',
       precedence: Precedences.ProperSolemnity_PrincipalPatron_4a,
       titles: { append: [PatronTitles.PatronOfCanada] },
     },
 
+    // src: mr_fr_2021_ed3
     kateri_tekakwitha_virgin: {
-      precedence: Precedences.OptionalMemorial_12,
+      precedence: Precedences.ProperMemorial_11b,
       dateDef: { month: 4, date: 17 },
     },
 
+    // src: mr_fr_2021_ed3
     marie_anne_blondin_virgin: {
       precedence: Precedences.OptionalMemorial_12,
       dateDef: { month: 4, date: 18 },
     },
 
+    // src: mr_fr_2021_ed3
     our_lady_of_good_counsel: {
       precedence: Precedences.OptionalMemorial_12,
       dateDef: { month: 4, date: 26 },
     },
 
+    // src: mr_fr_2021_ed3
     mary_of_the_incarnation_marie_guyart_religious: {
-      precedence: Precedences.OptionalMemorial_12,
+      precedence: Precedences.ProperMemorial_11b,
       dateDef: { month: 4, date: 30 },
     },
 
+    // src: mr_fr_2021_ed3
+    pius_v_pope: {
+      dateDef: { month: 5, date: 1 },
+    },
+
+    // src: mr_fr_2021_ed3
     marie_leonie_paradis_virgin: {
       precedence: Precedences.OptionalMemorial_12,
```

Comment:
I presume that optional memorial is the default celebration rank.

```suggestion
```

Reply:
There is currently no default value for the `precedence` property. This property, along with the `dateDef` property, are the bare essentials to define a calendar entry (unless it's a definition that extends a higher already defined definition)

Reply:
I also believe it's better to systematically and explicitly define this property (except when extending a previously defined definition). It's an important property in the calendar computing. It's also a way to make these definitions more readable and understandable for anyone.

FYI, if the `precedence` property is not defined on a calendar definition (as well as the `dateDef` property), an error is thrown during the compilation of Romcal.

Reply:
Okay, then ignore these removal suggestions.

I was under the impression that is is used default. However, I agree with you it is much better to explicitly define it (just like you described it).

Anything, when you’ll work on the ESLint plugin, you should add these to the checks:
- useless re-definition of same celebration properties (with the same value) when those are already inherited;
- missing required properties.

Reply:
> when you’ll work on the ESLint plugin, you should add these to the checks

💯